### PR TITLE
New cli tools

### PR DESCRIPTION
**Pull request by thijsmie**
This adds a new suite of command line tools to the python backend:

```sh
# list the entities discovered on the network
$ cyclonedds ls
# list the applications discovered on the network (once participant-properties are in)
$ cyclonedds ps
# discover and reconstruct the IDL of a topic type
$ cyclonedds typeof topic_name
# discover and subscribe to an arbitrary topic
$ cyclonedds subscribe topic_name
```

### CODE REVIEW

File: cyclonedds/tools/cli/discovery/main.py
Comment (eboasson) [inline cyclonedds/tools/cli/discovery/main.py:145]: "Unkown" is probably be a typo, and copied to the following three lines

**Review by eboasson — APPROVED** [review]
Neat! From what I can tell there's no hard dependency on https://github.com/eclipse-cyclonedds/cyclonedds/pull/1281, so no need to wait for it.

That means the only comments I have are a typo and wondering how the color scheme works out for people using a light background. I should try that out myself ... I know.

Neither is a reason to not approve this already. Thanks @thijsmie !

File: cyclonedds/tools/cli/discovery/main.py
Comment (thijsmie) [reply cyclonedds/tools/cli/discovery/main.py]: Good catch, fixed that.

**Comment by thijsmie** [comment]
Indeed there is no hard dependency on the participant-properties, given that we can't rely on them always being there anyways it would not be sensible to require them for the tool to work.

As for the color scheme, I have no idea! I have left basically all settings of the wonderful [click](https://github.com/pallets/click/) by @pallets, [rich](https://github.com/Textualize/rich) by @willmcgugan and [rich-click](https://github.com/ewels/rich-click) by @ewels libraries on default and I have good hope that that will 'just work'! I'll see if I can manage a testrun by fiddling with my terminal settings.

**Comment by eboasson** [comment]
@thijsmie, if `pip install --user --upgrade pip` now suddenly fails on Windows, I'd say it is unrelated to this PR. As the CI was fine on all platforms until I merged https://github.com/eclipse-cyclonedds/cyclonedds/pull/1281 and neither that nor the changes you made to this PR in response are platform dependent, I'd say the likelihood of this PR introducing any problems on Windows is negligible. As far as I am concerned, this can be merged right now.

**Comment by thijsmie** [comment]
Yeah I don't know what goes on with windows either, I will merge now and hopefully the problem with pip will go away, if not I'll see about changing the way the windows python environment is set up.

**Pull request merged**
